File: python/content_extraction.py
#!/usr/bin/python

import sys
import tidy

import CEviaTD;

if len(sys.argv) <= 2:
    print >> sys.stderr, "Usage: %s input_html output_html" % sys.argv[0]
    sys.exit(1)

input_html  = sys.argv[1]
output_html = sys.argv[2]

encoding = 'utf8'

input_content = ""
for line in open(input_html):
    input_content += line

tidy_options = {'char_encoding'    : encoding,
                'doctype'          : 'omit',
                'indent'           : 0,
                'numeric_entities' : 1,
                'output_xhtml'     : 1,
                'tidy_mark'        : 0,
                'wrap'             : 0}

trans_input = tidy.parseString(input_content, **tidy_options)

if len(str(trans_input)) == 0:
    print >> sys.stderr, "Fail to tidy html"
    print [e for e in trans_input.errors if e.severity == 'E']
    sys.exit(1)

trans_input = str(trans_input)

extractor = CEviaTD.CEviaTD();
ret = extractor.Parse(trans_input)

if ret < 0:
    print >> sys.sterr, "Fail to extract content"
    sys.exit(1)

extracted_xml  = extractor.GetExtractedXml()
extracted_text = extractor.GetExtractedText()

trans_output = tidy.parseString(extracted_xml, **tidy_options)

print extracted_text

f = open(output_html, "w")
f.write(extracted_xml)
f.close()

File: perl/content_extraction.pl
#/usr/local/bin/perl

use strict;
use warnings;

use Encode;
use HTML::Tidy;

use CEviaTD;

main();

sub main {
    if (@ARGV < 2) {
        print STDERR "Usage: $0 input_html output_html\n";
        exit 1;
    }
    my $input_html  = shift @ARGV;
    my $output_html = shift @ARGV;

    my $encoding = 'utf8';

    my $input_content = "";
    open(my $ifh, '<', $input_html) or die $!;
    while (my $line = <$ifh>) {
        $input_content .= $line;
    }
    close($ifh);

    my $tidy = HTML::Tidy->new({
            doctype          => 'omit',
            indent           => 0,
            numeric_entities => 1,
            output_xhtml     => 1,
            tidy_mark        => 0,
            wrap             => 0,
    });
    my $trans_input = $tidy->clean(decode($encoding, $input_content));

    my $extractor = CEviaTD::CEviaTD->new();
    my $ret = $extractor->Parse($trans_input);

    if ($ret < 0) {
        print STDERR "Failt to extract content\n";
        exit 1;
    }

    my $extracted_xml  = $extractor->GetExtractedXml();
    my $extracted_text = $extractor->GetExtractedText();

    my $trans_output = encode($encoding, $tidy->clean(decode($encoding, $extracted_xml)));

    print $extracted_text, "\n";

    open(my $ofh, '>', $output_html) or die $!;
    print {$ofh} $trans_output;
    close($ofh);
}
